theforeman.org/pipelines/release/pipelines/candlepin.groovy

```diff
@@ -0,0 +1,67 @@
+pipeline {
+    agent none
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('staging-build-repository') {
+            steps {
+                git url: "https://github.com/theforeman/theforeman-rel-eng", poll: false
+
+                script {
+                    candlepin_distros.each { distro ->
+                        sh "./build_stage_repository candlepin ${candlepin_version} ${distro}"
+                    }
+                }
+            }
+        }
+        stage('staging-copy-repository') {
+            steps {
+                script {
+                    dir('tmp') {
+                        rsync_to_yum_stage('candlepin', 'candlepin', candlepin_version)
+                    }
+                }
+            }
+        }
+        stage('staging-repoclosure') {
+            steps {
+                script {
+                    def parallelStagesMap = [:]
+                    def name = 'candlepin-staging'
+                    candlepin_distros.each { distro ->
+                        parallelStagesMap[distro] = { repoclosure(name, distro, candlepin_version) }
+                    }
+                    parallel parallelStagesMap
+                }
+            }
+            post {
+                always {
+                    deleteDir()
+                }
+            }
+        }
+        stage('staging-push-rpms') {
+            agent { label 'admin && sshkey' }
+
+            steps {
+                script {
+                    candlepin_distros.each { distro ->
+                        //push_foreman_staging_rpms('candlepin', candlepin_version, distro)
+                        echo "Push RPMs to production disabled"
```

Comment:
This is intentional while I iterate on this, I'd like to have the staging part available.